rfcs/0052-create-sink-into-table.md

```diff
@@ -161,7 +150,82 @@ GROUP BY window_start;
 ```
 
 ### SQL evolution
-This case is very similar to some AP system's conditional-updates behavior. https://docs.starrocks.io/en-us/latest/loading/Load_to_Primary_Key_tables#conditional-updates
+After decoupling the sink and the table. Users have chance to change the sink's SQL by drop the old sink and rebuild the new sink to the table. In that case, the table's downstream will not be affected. 
+But RW can not give consistent guarantee, so user must give additional declaration in SQL.
+
+I summarize two patterns here
+- Backfill: If user have all historical data in the upstream table, he can drop the old the create the new sink to the table
+  - 👍 Easy to understand. No more SQL rewritten
+  - 👍 eventually consistency
+  - 👎 The result table and all its down stream mv's version will rollback and seems a stale read for user.
+  - 👎 Long time to backfill and re-compute all the data.
+  - 👎 In some cases, user can not get all upstream historical data such as a kafka source with TTL.  
+- Apply delta: User can determine a timestamp of event time to determine use the old sink or the new sink to update the result table.
+  - 👍 No result rollback
+  - 👍 Incremental computation
+  - 👎 Limited with event time column in the result table.
+  - 👎 Might be complex for user to write on conflict.
+
+Examples :
+1. This is a Backfill case but user define a proper on conflict behavior with the monotonically increasing `order_cnt` to prevent data rollback. 
+```SQL
+  CREATE TABLE dwd_orders (
+    user VARCHAR,
+    product VARCHAR,
+    price BIGINT,
+    event_time TIMESTAMP,
+) APPEND ONLY;
+
+CREATE TABLE dws_product (
+  product VARCHAR,
+  order_cnt BIGINT,
+  PRIMARY KEY (product)
+) WITH (
+  'ttl' = '7d'
+  -- only update when the new incoming row's order_cnt is larger than the old order_cnt
+) ON CONFLICT DO UPDATE SET (product, order_cnt) = (
+      CASE WHEN EXCLUDED.order_cnt IS NOT NULL AND EXCLUDED.order_cnt > order_cnt 
+        THEN EXCLUDED.product ELSE product 
+      END,
+      CASE WHEN EXCLUDED.order_cnt IS NOT NULL AND EXCLUDED.order_cnt > order_cnt 
+        THEN EXCLUDED.order_cnt, order_cnt ELSE order_cnt 
+      END)
+
+CREATE SINK order_production_analyze_job INTO dws_product AS 
+SELECT 
+  window_start,
+  product,
+  COUNT(*) as order_cnt,
+FROM dwd_orders
+GROUP BY product
+WITH (
+  -- because the table's on- conflict behavior can only applied on the “INSERT” conflict，so we need use force append-only sink here.
+  force_append_only = 'true'
+);
+
+-- when the user want to add an new aggregator
```

Comment:
I don't really like this use case. It basically adds another identical streaming job (near 2x cost) just to put a new aggregate function there. @BugenZhao's design (TODO: not published to RFC repo yet) would be a better approach.